nav2_bringup/bringup/launch/multi_tb3_simulation_launch.py

```diff
@@ -100,26 +100,12 @@ def generate_launch_description():
         default_value='True',
         description='Whether to start RVIZ')
 
-    # Start Gazebo with plugin providing the robot spawing service
+    # Start Gazebo with plugin providing the robot spawning service
     start_gazebo_cmd = ExecuteProcess(
-        cmd=[simulator, '--verbose', '-s', 'libgazebo_ros_factory.so', world],
+        cmd=[simulator, '--verbose', '-s', 'libgazebo_ros_init.so',
+                                     '-s', 'libgazebo_ros_factory.so', world],
         output='screen')
 
-    # Define commands for spawing the robots into Gazebo
-    spawn_robots_cmds = []
```

Comment:
This is the multi-robot launch file, why was this all deleted? Don't we need need something to spawn the 2 different robots?

Reply:
I see you're forwarding info below, but if we spawned all `N` robots here, couldn't we just have their navigation systems and 1 simulator launch in the other file?

I suppose I just don't understand why changes needed to be made structurally so that we could just separate out the world and robot model files. Could the pose be read in as a array vector rather than independent X, Y,and Z components? Because we also need to consider orientation as well.

Reply:
sorry i didn't  get notification to reply quickly

the pose can be read as vector if you suggest.

mutli_tb3_simulation_launch.py  file uses tb3_simulation_launch.py https://github.com/ros-planning/navigation2/blob/792893a5b56256cff8e9d4fb581e71d23892b5d6/nav2_bringup/bringup/launch/multi_tb3_simulation_launch.py#L142
before there is no nav2_gazebo_spawner_node in tb3_simulation_launch file.  in order to seperate  world and robot nav2_gazebo_spawner _node  should be created in tb3_simulation_launch file. i thought its neat to use this node  in multi_tb3 file as launch configuration input.
  we  can also  remove the file [spawn_tb3_launch.py](https://github.com/ros-planning/navigation2/blob/main/nav2_bringup/bringup/launch/spawn_tb3_launch.py) since it contains a single node .

@SteveMacenski  if you prefer the earlier structure . the node can be ignored with if condition (simulator) and the structure can be restored.

Reply:
Let me think about this (aloud), these are just be spitballing ideas here. I don't know what the best solution is, but lets chat it over and use the ones we like the best. I don't think there's an objective right answer to this one

- We do what you suggest in the PR now, have `N` of the Tb3 launch files called each spawning its own robot with simulation set to false. We should add a comment in the code in this case that the spawning is happening when navigation/simulation is coming up.
- We have an option in the Tb3 sim file for whether or not to spawn a robot, set true default and use false for the multirobot case and have that multirobot launch file use the spawner in a for loop 

Of those two options, I like what you have now better. What do you think?

---

A separate topic, but what if we did all the finding robot SDF file stuff in the spawner launch file to abstract that from these launch files (e.g. https://github.com/ros-planning/navigation2/pull/2473/files#diff-f2cb98e53f28476ba70befa88ddc8279de87edf8b4bd6de915f1dbd844281252R182-R183)? That way that launch file is self contained in finding and spawning the robot (probably with a launch configuration to override).

Reply:
i am inclined to first option thats why i removed spawn block in multi_tb3_launch file. now the single tb3_simulation_launch file has two launch description includes bringup for nav2 related nodes. rviz to start rviz everything else in it is specific to robot.
i think its more user friendly to beginners to try change robot urdf, sdf, world, map and  simulator
 multi _launch file just starts gazebo and iterates for N robots.

about the separate topic why create separate launch file for spawning only how about we club simulator specific nodes or robot + simulator +robot_state _publisher node since if one change urdf, sdf  also need to be changed. also it would be more convenient to change simulator like ignition , webot etc

just my thoughts

Reply:
converting the pose as a vector in launch argument is not straight forward. i am finding  how to do it. do you came across any reference?

Reply:
OK. 

On the vector question, I think it would be good to look at the TF static broadcaster as an example, they take in the exact same kind of vector we're talking about.